docs/intro/overview.rst

```diff
@@ -13,172 +13,112 @@ precisely, `web scraping`_), it can also be used to extract data using APIs
+Spider definition inside it and ran it through its crawler engine.
 
+The crawl started by making requests to the URLs defined in the ``start_urls``
+attribute (in this case, only the URL for StackOverflow top questions page),
+and called the default callback method ``parse`` passing the response object as
+an argument. In the ``parse`` callback, we extract the links to the
+question pages using a CSS Selector with a custom extension that allows to get
+the value for an attribute. Then, we yield a few more requests to be sent,
+registering the method ``parse_question`` as the callback to be called for each
+of them as they finish.
 
-.. highlight:: none
+Here you notice one of the main advantages about Scrapy: requests are
+:ref:`scheduled and processed asynchronously <topics-architecture>`.  This
+means that Scrapy doesn't need to wait for a request to be finished and
+processed, it can send another request or do other things in the meantime. This
+also means that other requests can keep going even if some request fails or an
+error happens while handling it.
 
-An XPath expression to select the file size could be::
+While this enables you to do very fast crawlings sending multiple concurrent
+requests at the same time in a fault-tolerant way, Scrapy also gives you
+control over the politeness of the crawl through :ref:`a few settings
+<topics-settings-ref>`. You can do things like setting a download delay between
+each request, limit amount of concurrent requests per domain or per IP, and
+even :ref:`use an auto-throttling extension <topics-autothrottle>` that tries
+to figure out these automatically.
 
-   //div[@id='specifications']/p[2]/text()[2]
+Finally, the ``parse_question`` callback scrapes the question data for each
+page yielding a dict, which Scrapy then collects and writes to a JSON file as
+requested in the command line.
 
-.. highlight:: python
+.. note::
 
-For more information about XPath see the `XPath reference`_.
+    This is using :ref:`feed exports <topics-feed-exports>` to generate the
+    JSON file, you can easily change the export format (XML or CSV, for example) or the
+    storage backend (FTP or `Amazon S3`_, for example).  You can also write an
+    :ref:`item pipeline <topics-item-pipeline>` to store the items in a database.
 
-Finally, here's the spider code::
+The data in the file will look like this (note: formatted for easier reading)::
 
-    from scrapy.contrib.spiders import CrawlSpider, Rule
-    from scrapy.contrib.linkextractors import LinkExtractor
 
-    class MininovaSpider(CrawlSpider):
+    [{
+            "body": "... LONG HTML HERE ...",
+            "link": "http://stackoverflow.com/questions/11227809/why-is-processing-a-sorted-array-faster-than-an-unsorted-array",
+            "tags": ["java", "c++", "performance", "optimization"],
+            "title": "Why is processing a sorted array faster than an unsorted array?",
+            "votes": "9924"
+        },
```

Comment:
indentation of open `{` brace doesn't match indentation of closing `}` brace - is it intentional?